From: Ben Hutchings <ben@decadent.org.uk>
Subject: ethernet: define eth_hw_add_random()
Date: Mon, 28 Jan 2013 00:03:32 +0000
Bug-Debian: http://bugs.debian.org/699129
Forwarded: not-needed

This was defined in commit 1a0d6ae5795c376bae6d012fb25e8341e4c6d5f2
upstream, where it replaced dev_hw_addr_random().  Obviously we
can't remove the latter.

---
--- a/include/linux/etherdevice.h
+++ b/include/linux/etherdevice.h
@@ -154,6 +154,21 @@ static inline void dev_hw_addr_random(st
 }
 
 /**
+ * eth_hw_addr_random - Generate software assigned random Ethernet and
+ * set device flag
+ * @dev: pointer to net_device structure
+ *
+ * Generate a random Ethernet address (MAC) to be used by a net device
+ * and set addr_assign_type so the state can be read by sysfs and be
+ * used by userspace.
+ */
+static inline void eth_hw_addr_random(struct net_device *dev)
+{
+	dev->addr_assign_type |= NET_ADDR_RANDOM;
+	random_ether_addr(dev->dev_addr);
+}
+
+/**
  * compare_ether_addr - Compare two Ethernet addresses
  * @addr1: Pointer to a six-byte array containing the Ethernet address
  * @addr2: Pointer other six-byte array containing the Ethernet address
